[s] Welcome to Nova Edina!

Thalpy opened the pull request:
## About The Pull Request

It's festive time!

## Why It's Good For The Game

_Holiday cheer!_

## Changelog
:cl:
add: Adds FestiveBall! (and a ton of backup maps) 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
Added Fitz's maps

Set up branch
Added 2 Fitz .dmm files

Commit:
Adds a chem example

Yey

Commit:
Updates FermiExample

Commit:
Small fix

Commit:
Cobblestone & Sidewalk Turfs w/ smoothing

Adds both cobblestone road, and sidewalks with smoothing with already done coding so you can get right into drawing up the roads and sidewalks.

Commit:
Adds open & closed turfs + decals and a sample layout map

Layout idea, compile and check it out. Adds alleyway turfs like bricks, no variation yet to dirtiness or randomness like missing a brick. Also adds rails which can be used to detail where a trail would be and grey brick walling with smoothing functionality.

Commit:
Adds a bunch of stuff

3 new brick types, grey bricks now have a window object you can place. Added festives lamps, three variants.

Commit:
Enables dm

Commit:
Enables 32x64 icons.

Lamp

Commit:
Fiddling with ideaas

Commit:
More hospital

Commit:
Updater

Commit:
Bricks, windows, and street decals.

Commit:
Fixes brick windows to be windows

Forgot to add anchor & density to them.

Commit:
My changes

Commit:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Commit:
Adds the hospital

Commit:
Remember to puush

Commit:
Few size things

Commit:
Okay I'm seriously stopping

ThalpySci committed:
Merge branch 'cargo' into FestiveBall

EmeraldSundisk committed:
Update chrimbus2_lofat.dmm

EmeraldSundisk committed:
Merge pull request #3 from FestiveBall/combination

Prepares the Syndicate base for integration.

ThalpySci committed:
Adds vectorcrafts and a few sprites

Commit:
Fixes crashing from loading extra maps

Commit:
I don't know

Commit:
HURGH

Commit:
I'mma fudge it

Commit:
SPEEDYBOI.. as they say

Commit:
Adds 8 direction

Commit:
Train platform + window color types extended w/ small other things

Commit:
More train platform stuff

EmeraldSundisk committed:
So here's the library.

EmeraldSundisk committed:
And now it's updated a little.

Commit:
Ready for testing

Commit:
Updates cars

Commit:
Fix?

Commit:
Adds two wooden tile types, and a clinic.

Commit:
Fixes map, I hope

Commit:
Expands/Edits roads

Commit:
Vectorcars semifunctional

Commit:
Vectorcars function complete

Commit:
Okay I'm done

Commit:
Tweaks and stuff

Commit:
Adds tint buttons

Commit:
Naps

Commit:
Fixed!

Commit:
Fixesssss

HARK

Commit:
How to

Commit:
Okay I'm done fiddling

PersianXerxes committed:
Merge pull request #6 from FestiveBall/master

Updating branch to match Fermi.

PersianXerxes committed:
Merge pull request #7 from FestiveBall/FestiveBall

Festive ball update

Commit:
Little bit of work on the map

PersianXerxes committed:
Unfinished map changes, added the start of Science.

PersianXerxes committed:
Working around a merge conflict.

PersianXerxes committed:
Bypassing merge conflict

PersianXerxes committed:
Merge pull request #6 from PersianXerxes/christmasMapping

Christmas mapping (Merge conflict bypassed)

KathrinBailey committed:
okay I have to do real life stuff I started on a mockup for a bar it obviously isn't done

very not done please mostly ignore

Commit:
More map edits, fixed some space tiles.

Commit:
Bunch of stuff

PersianXerxes committed:
Merge pull request #8 from FestiveBall/FestiveBall

Updating Festive ball

Commit:
Adds cream bricks

Commit:
Fixes Persi's map, retrieved Fitz maps which were deleted.

Commit:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

PersianXerxes committed:
Merge pull request #9 from FestiveBall/FestiveBall

Adds cream bricks

PersianXerxes committed:
Merge pull request #10 from FestiveBall/FestiveBall

Festive ball UPDATE

PersianXerxes committed:
Updates to the map.

PersianXerxes committed:
Merge pull request #7 from PersianXerxes/christmasMapping

Science Expanded

Commit:
SItches everything so far together

Commit:
Wires and atmos medbay

Commit:
Smol fixes

Commit:
Okay fixed (don't forget colison)

KathrinBailey committed:
Bar is getting along, needs to be merged with stuff.

KathrinBailey committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

KathrinBailey committed:
More progress on the pub!

Fermis's edits and mine are on the same map now.

EmeraldSundisk committed:
GET A LOAD OF THIS.

Thalpy committed:
Don't edit tge

Commit:
HECK

Commit:
Merge branch 'emerald' into FestiveBall

Commit:
Slowly integrating

PersianXerxes committed:
Merge pull request #11 from FestiveBall/FestiveBall

Festive ball

KathrinBailey committed:
KAT STATION

I WANNA COPY AND PASTE MY OWN STUFF AND NOT SWITCH DMEs CONSTANTLY!

Commit:
Okay sorted out most things

Okay sorted out most things

Commit:
Updates

Commit:
BIG SLEEPY FERMI

PersianXerxes committed:
Merge pull request #12 from FestiveBall/FestiveBall

Festive ball

PersianXerxes committed:
Expands Science slightly STILL NOT FINISHED, adds Command ALSO UNFINISHED

PersianXerxes committed:
Merge pull request #10 from PersianXerxes/christmasMapping

Xenobio, start of Nanites, and Command

Commit:
Added plumbing

Commit:
Wires it up

Commit:
trains

Commit:
Okay I'm stopping for now

Commit:
OOPS

PersianXerxes committed:
Merge pull request #13 from FestiveBall/FestiveBall

Festive ball

PersianXerxes committed:
Finished Science layout, decorations remain.

PersianXerxes committed:
Merge pull request #11 from PersianXerxes/christmasMapping

Christmas mapping

Commit:
Cargooooooo

Commit:
Cargoooooooooooooo

Commit:
Cargo done?

Commit:
Rough plan

Commit:
Pushhh

Commit:
I just can't stop fiddling

Commit:
Rough cargo done

Commit:
Adds custom sprites for heads

Commit:
Updates cars

KathrinBailey committed:
I fell asleep like a moron and other people were working on it.

I'm such a stupid loser.

KathrinBailey committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Commit:
Slowly getting there

KathrinBailey committed:
Service is feature complete now it just needs being complete complete

cat floofhat!

KathrinBailey committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

KathrinBailey committed:
food cart

meow meow floofcat!

KathrinBailey committed:
PEW PORT PEW PORT MAY NEED REVERT PEW PORT

MAY BREAK EVERYTHING PEW PORT

KathrinBailey committed:
Revert "PEW PORT PEW PORT MAY NEED REVERT PEW PORT"

This reverts commit 3f03e5accafa9fc88424b878106c0ef673e21871.

KathrinBailey committed:
Revert "Revert "PEW PORT PEW PORT MAY NEED REVERT PEW PORT""

This reverts commit 78d86b3bb8de451a9ba2fa5c9d9565f8656e60a3.

KathrinBailey committed:
Pews should work?

It worked in-game.

EmeraldSundisk committed:
Update GabbyLayout.dmm

Thalpy committed:
Merge pull request #18 from FestiveBall/emeraldagain

Your friendly neighborhood area designator.

EmeraldSundisk committed:
Update GabbyLayout.dmm

Thalpy committed:
Merge pull request #20 from FestiveBall/yetanotherbranch

Adds an arcade, along with a small chapel fix.

BlackMajor committed:
Started on the sec region

ThalpySci committed:
Adds edina areas

ThalpySci committed:
Compiles

ThalpySci committed:
Changes area types

Thalpy committed:
Area tweaks

ThalpySci committed:
Updates vectorcrafts

ThalpySci committed:
Commit

ThalpySci committed:
oops breaks everything

ThalpySci committed:
Fixes compile errors

ThalpySci committed:
Few things

ThalpySci committed:
SLeepytimes

ThalpySci committed:
oops

ThalpySci committed:
HELLO IT ME A BOOKSMITH

ThalpySci committed:
Update Sleeper.dm

sneaky runtimes

BlackMajor committed:
Unga sec area work

BlackMajor committed:
More sec work

BlackMajor committed:
tiny update

Thalpy committed:
Update vehicles.dmi

Thalpy committed:
Kitten mittens

Thalpy committed:
Oops compile

Thalpy committed:
Few changes

ThalpySci committed:
Brighten up those lamps!

EvaMistral committed:
Atmosia shell

Added a basic shell for Atmospherics, Added gas storage tanks, Added Pure pipeline, Layer manifolds on all tanks except Oxygen and Nitrogen not working - To fix, Pipe missing under Air mixer, Excessive lights everywhere to be fixed, Grav generator to be fixed, Cameras to be added, Wires to be added, temp APC needs moved and wired. gas miners are sexy

EvaMistral committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

EvaMistral committed:
some Commit stuff!

Hopefully i didn't break anything. but progress is progress

BlackMajor committed:
Some more sec work to spruce it up

PersianXerxes committed:
Merge pull request #14 from FestiveBall/FestiveBall

Updating to finish my parts of the map, sorry for the wait

PersianXerxes committed:
Wrapping up stuff, added berets

PersianXerxes committed:
Merge pull request #23 from PersianXerxes/christmasMapping

Further Science and Command Work

BlackMajor committed:
Finishing sec

Thalpy committed:
Big changes reduces map size

Thalpy committed:
Okay stopping for now

Thalpy committed:
Adds a gutted spookystation that I can copy/paste from

EvaMistral committed:
Basic Piping to Sci

Adding basic HE pipes, still needs linked up

EvaMistral committed:
Pushing for small piping additions to science again

Added a few more pipes that took like 5 minutes, totally worth another push : 3

Useroth committed:
Added strip club.

Useroth committed:
Turns out we don't use dmm2tgm here.

BlackMajor committed:
Map work and sidewalk fix

BlackMajor committed:
Pushing what I have while busy

BlackMajor committed:
Should probably mark the space im gonna use

KathrinBailey committed:
Tiny Kat before-I-go-to-work additions

ThalpySci committed:
Sleepy coding on the train

ThalpySci committed:
scarves and decals

BlackMajor committed:
Work on AI core

KathrinBailey committed:
Kat dorms addition ahhh I'm going to be late for work

KathrinBailey committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

KathrinBailey committed:
Merged rushed as fuck dorms oof

Aaahh sorry it would've been better if I had more time fuck these morning shifts.

ThalpySci committed:
Tweaking NOT DONE

ThalpySci committed:
cyber

ThalpySci committed:
More edits

ThalpySci committed:
HURGH

Thalpy committed:
HECK

Thalpy committed:
Done

Thalpy committed:
Alpha map

Thalpy committed:
Fixes a whole buncha stuff

Thalpy committed:
Merge remote-tracking branch 'upstream/master' into FestiveBall

Thalpy committed:
Updates car crashes

EmeraldSundisk committed:
Whoops.

EmeraldSundisk committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Thalpy committed:
Cleans up maps

Thalpy committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Thalpy committed:
Undoes stuff I hope

Thalpy committed:
Revert "Undoes stuff I hope"

This reverts commit d5bbe92f2c67d708c064e5c911be5f182a56d047.

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+	icon_state = "edina"
+	valid_territory = FALSE
+	has_gravity = STANDARD_GRAVITY
+	icon = 'icons/turf/areas.dmi'
```
Same as base area type. Remove.

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+	power_environ = FALSE
+	valid_territory = FALSE
+	outdoors = TRUE
+	ambientsounds = SPACE
```
I doubt spess ambients sounds would embody well the snowy yet festive ambient of the map.

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+area/edina
+	name = "Nova Edina wilderness"
+	icon_state = "edina"
+	valid_territory = FALSE
```
Already defined on line 16.

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+	icon = 'icons/turf/areas.dmi'
+	clockwork_warp_allowed = FALSE // Can servants warp into this area from Reebe?
+	clockwork_warp_fail = "The aurora borealis is interfering with your teleport! Try somewhere closer to the city."
+	requires_power = TRUE
```
Same as base.

Thalpy committed:
Fixes DME?

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+	light_range = 3 //MIDNIGHT BLUE
+	light_power = 0.15 //NOT PITCH BLACK, JUST REALLY DARK
+	light_color = "#00111a" //The light can technically cycle on a timer worldwide, but no daynight cycle.
+	baseturfs = /turf/open/floor/grass/snow/edina //If we explode or die somehow, we just make more! Ahahaha!!!
```
Uuuh, this makes me want to self-merge the mapping code update so you can just use the new ZTRAIT_BASETURF mapping trait instead. Really.
Otherwise good lord I hope you have set the deprecable base_turf helper for each area.

Thalpy replied:
I'll tell you what,I'll promote you to head baseturfer for the final non festive variant of the map.

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+	blob_allowed = FALSE //Eating up space doesn't count for victory as a blob.
+
+	/// If false, loading multiple maps with this area type will create multiple instances.
+	unique = TRUE
```
Same as base.

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+
+/////////////////Edina specific derivitives///////////////////////////////////
+
+area/edina/crew_quarters/holo_atrium
```
There is no base `/area/edina/crew_quarters` that allows clockies, blobs or gangsters.... Oh well this is a festive map so the gamemode will be extended anyway.

Thalpy replied:
fixed

Ghommie commented on code/game/area/areas/edina.dm:
```diff
@@ -0,0 +1,84 @@
+area/edina
```
Oh yea, add an initial / to each typepath that doesn't have one already.

Ghommie commented on code/modules/clothing/gloves/mittens.dm:
```diff
@@ -0,0 +1,29 @@
+
+/obj/item/clothing/gloves/mittens/random
+
+/obj/item/clothing/gloves/mittens/random/New()
```
Use Initialize() and add a parent call right after or the item won't be properly initialized, unpickable and other bad things.

Ghommie commented on code/modules/clothing/gloves/mittens.dm:
```diff
@@ -0,0 +1,29 @@
+	if(picked_c == "kitten")
+		new /obj/item/clothing/gloves/mittens/kitten(loc)
+		qdel(src)
+		return
```
```suggestion
		return INITIALIZE_HINT_QDEL
```
Initialize stuff.

Ghommie commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -445,6 +445,19 @@
+	var/mob/living/carbon/human/H = owner
+	if(!H)
+		return FALSE
+	if(!H.wear_neck)
+		return FALSE
+	if(istype(H.wear_neck, /obj/item/clothing/neck/scarf))
+		return TRUE
+	var/striped_scarves = list(/obj/item/clothing/neck/stripedredscarf, /obj/item/clothing/neck/stripedgreenscarf, /obj/item/clothing/neck/stripedbluescarf)
+	if(is_type_in_list(H.wear_neck, striped_scarves))
+		return TRUE
+	return FALSE
```
All of these typechecks could be replaced with a clothing flag and then this run instead. More optimal code.
```
	for(var/obj/item/clothing/C in list(H.wear_neck, H.wear_mask, H.head))
		if(C.clothing_flags & PROTECT_COLD_AIR)
			return TRUE
	return FALSE
```

Thalpy replied:
I'll do this in the future

EmeraldSundisk committed:
Update FestiveStation.dmm

Thalpy committed:
Fixes

Ghommie commented on modular_citadel/code/modules/festive/vector_process.dm:
```diff
@@ -0,0 +1,28 @@
+PROCESSING_SUBSYSTEM_DEF(vectorcraft)
+	name = "Vectorcraft Movement"
+	priority = 40
```
Add a new fire_priority define for this, pretty please.

Ghommie commented on modular_citadel/code/modules/festive/turfs.dm:
```diff
@@ -0,0 +1,1671 @@
+	else
+		..(FULLTILE_WINDOW_DIR)
+
+/obj/structure/festive/trainplatform/edge_north/CanPass(atom/movable/mover, turf/target)
```
I swear, onborder objects' CanPass code should be refactored one day, just saying.

Ghommie commented on modular_citadel/code/modules/festive/turfs.dm:
```diff
@@ -0,0 +1,1671 @@
+	else
+		..(FULLTILE_WINDOW_DIR)
+
+/obj/structure/festive/trainplatform/edge_east/CanPass(atom/movable/mover, turf/target)
```
Oh man, at this point you could have just done a variable and brought all of this CanPass / CheckExit / SetDir / Initialize overrides down to the base trainplatform. WTH?! :sweat_smile:

Ghommie commented on modular_citadel/code/modules/festive/vector_process.dm:
```diff
@@ -0,0 +1,28 @@
+	stat_tag = "VC"
+	flags = SS_NO_INIT|SS_TICKER|SS_KEEP_TIMING
+
+	//var/flightsuit_processing = FLIGHTSUIT_PROCESSING_FULL
+
+
+/* unsure if needed
+/datum/controller/subsystem/processing/vectorcraft/Initialize()
+	sync_flightsuit_processing()
+
+/datum/controller/subsystem/processing/vectorcraft/vv_edit_var(var_name, var_value)
+	..()
+	switch(var_name)
+		if("flightsuit_processing")
+			sync_flightsuit_processing()
+
+/datum/controller/subsystem/processing/vectorcraft/proc/sync_flightsuit_processing()
+	for(var/obj/vehicle/sealed/vectorcraft/VC in processing)
+		VC.sync_processing(src)
+	if(flightsuit_processing == FLIGHTSUIT_PROCESSING_NONE)	//Don't even bother firing.
+		can_fire = FALSE
+	else
+		can_fire = TRUE
+*/
```
Lotsa unused flightsuit inspired relic code.

Thalpy replied:
Yep, I'll clean it up for the final merge, it was mostly there if things went wrong

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,581 @@
+	.=..()
+	driver.client.pixel_x = 0
+	driver.client.pixel_y = 0
+	driver.pixel_x = 0
+	driver.pixel_y = 0
```
Why resetting the driver's pixel displacement too? This is a sealed vehicle subtype and the driver shouldn't even be visible... unless...

Thalpy replied:
Because I tried to get it working with wheelchairs

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,581 @@
+	*/
+
+//Passive hover drift
+/obj/vehicle/sealed/vectorcraft/proc/hover_loop()
```
Is this proc only called on process(), right? If so move it to process() itself and avoid some useless proc overhead.

Thalpy replied:
I like keeping process clean, as well as having the ability to call this proc, if I need to for whatever reason. Though, I'll reread it over later

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,581 @@
+//
+/obj/vehicle/sealed/vectorcraft/Bump(atom/M)
+	var/speed = calc_speed()
+	if(iscarbon(M))
```
Umh, this could work just fine with all living mobs, just make sure they are not anchored.

Thalpy replied:
Added a !anchorhed

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,581 @@
+		C.throw_at(throw_target, 10, 14)
+		C.adjustBruteLoss(speed/10)
+		to_chat(C, "<span class='warning'><b>You are hit by the [src]!</b></span>")
+		to_chat(driver, "<span class='warning'><b>You just ran into [C] you crazy lunatic!</b></span>")
```
And perhaps move the `throw_at` and `adjustBruteloss` lines below the chat messages since a frew mobs can end up being deleted from lethal damage.

Thalpy committed:
More lamps

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,581 @@
+	if(istype(M, /obj/))
+		var/obj/O = M
+		O.take_damage(speed*2.5)
+	..()
```
Any issue if we do `return ..()` instead

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+//Repairing
+/obj/vehicle/sealed/vectorcraft/attackby(obj/item/O, mob/user, params)
+	.=..()
+	if(istype(O, /obj/item/weldingtool) && user.a_intent != INTENT_HARM)
+		if(obj_integrity < max_integrity)
+			if(!O.tool_start_check(user, amount=0))
+				return
+
+			user.visible_message("[user] begins repairing [src].", \
+				"<span class='notice'>You begin repairing [src]...</span>", \
+				"<span class='italics'>You hear welding.</span>")
+
+			if(O.use_tool(src, user, 40, volume=50))
+				to_chat(user, "<span class='notice'>You repair [src].</span>")
+				apply_damage(-max_integrity)
+		else
+			to_chat(user, "<span class='notice'>[src] does not need repairs.</span>")
```
All of this could be moved to `welder_act(mob/living/user, obj/item/I)`.

Thalpy replied:
Will do

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+			to_chat(user, "<span class='notice'>[src] does not need repairs.</span>")
+
+//Heals/damages the car
+/obj/vehicle/sealed/vectorcraft/proc/apply_damage(damage)
```
I.. really won't bother you about "unsnowflakeying" this function right now, but keep in mind about the existence of `take_damage()` and `obj_break()` for later.

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+	//	angle += 360
+	//message_admins("x:[x], y: [y], angle:[angle]")
+
+	//I WISH I HAD RADIANSSSSSSSSSS
```
the TORADIANS macro exists btw.

Thalpy replied:
oof

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+//Converts "gear" from intent to numerics
+/obj/vehicle/sealed/vectorcraft/proc/convert_gear()
+	switch(gear)
+		if("help")
```
```suggestion
		if(INTENT_HELP)
```

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+	switch(gear)
+		if("help")
+			return 1
+		if("disarm")
```
```suggestion
		if(INTENT_DISARM)
```

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+			return 1
+		if("disarm")
+			return 2
+		if("grab")
```
```suggestion
		if(INTENT_GRAB)
```

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+			return 2
+		if("grab")
+			return 3
+		if("harm")
```
```suggestion
		if(INTENT_HARM)
```

Ghommie commented on modular_citadel/code/modules/festive/vectorcraft.dm:
```diff
@@ -0,0 +1,585 @@
+/obj/vehicle/sealed/vectorcraft/mob_enter(mob/living/M)
+	if(!driver)
+		driver = M
+		if(gear != "auto")
```
"auto" should be a define specific for this file.

Ghommie commented on modular_citadel/code/modules/festive/vectorvariants.dm:
```diff
@@ -0,0 +1,45 @@
+/obj/vehicle/sealed/vectorcraft/cyber
+	name = "Cyberscuttle"
```
```suggestion
	name = "Cybershuttle"
```
Unless the mispelling was intentional...

Thalpy replied:
It scuttles! I couldn't think of another name for it's inspirer

Ghommie commented on modular_citadel/code/modules/festive/vectorvariants.dm:
```diff
@@ -0,0 +1,45 @@
+/obj/vehicle/sealed/vectorcraft/cyber
+	name = "Cyberscuttle"
+	desc = "A rhobust hovercar that, until recently, could self drive. Unfortunately it was found that the AI was terrified of orbs and drove away from them at all costs. Overall a smashing hovercar."
```
```suggestion
	desc = "A robust hovercar that, until recently, could self drive. Unfortunately it was found that the AI was terrified of orbs and drove away from them at all costs. Overall a smashing hovercar."
```

Ghommie commented on modular_citadel/code/modules/festive/wheelchair.dm:
```diff
@@ -0,0 +1,114 @@
+	driver.pixel_x = pixel_x
+	driver.pixel_y = pixel_y
+	driver.forceMove(src.loc)
+	driver.dir = dir
```
```suggestion
	driver.setDir(dir)
```

Ghommie reviewed (commented):
Ok, a little review concerning a few code quality issues (since you probably want to actually add vectorcrafts in a separate PR later).
But for what it adds. It's a pretty nice PR. (Did not bother reviewing the sprite assets)

Thalpy committed:
Update modular_citadel/code/modules/festive/vectorvariants.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
More lights

Thalpy committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Thalpy committed:
Delete turf

Thalpy committed:
Revert "Delete turf"

This reverts commit 3140de4c71a547ee9b81276c6906f4f732a7ea47.

Thalpy committed:
How did this happen

Thalpy committed:
Update code/modules/clothing/gloves/mittens.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
Update modular_citadel/code/modules/festive/vectorcraft.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
Update modular_citadel/code/modules/festive/vectorcraft.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
Update modular_citadel/code/modules/festive/wheelchair.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
Update modular_citadel/code/modules/festive/vectorcraft.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
Update modular_citadel/code/modules/festive/vectorcraft.dm

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Thalpy committed:
HECK

Thalpy committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Thalpy committed:
Fixada code

Thalpy committed:
BONK

EmeraldSundisk committed:
Update FestiveStation.dmm

EmeraldSundisk committed:
Merge branch 'FestiveBall' of https://github.com/FestiveBall/Citadel-Station-13 into FestiveBall

Thalpy committed:
Tweaks

Thalpy committed:
Final edits and extra festive decals

Thalpy committed:
Few tweaks to cars so they're a bit more responsive.

Thalpy committed:
Few things for cars

Thalpy committed:
Makes trucks have less suck

EmeraldSundisk committed:
Update FestiveStation.dmm

Commit:
As much change as I could

Moved Botany and remade it
Finished certain buildings which were never finished
Added my street decals to spurce up look at bland endless roads
Added another hotel near pub.
Renamed a couple door names to show what it is, I encourage you also add names to your doors.
Moved cryogenics near the starting station, it's labeled and roofed.
Made another janitors closet, preferrably remove the one up north east.
More work next couple pushes.

PersianXerxes committed:
Merge pull request #15 from FestiveBall/FestiveBall

Festive ball

ThalpySci committed:
Updates driving message

PersianXerxes committed:
MOST DORMS HAVE BUTTONS NOW

PersianXerxes committed:
Merge pull request #24 from PersianXerxes/christmasMapping

Christmas mapping

EmeraldSundisk committed:
Update FestiveStation.dmm

Thalpy committed:
Maptests

Thalpy committed:
Update StreetMockup.dmm
